src/orion/algo/hyperband.py

```diff
@@ -0,0 +1,423 @@
+        for i in range(0, num_brackets - bracket_id + 1):
+            n_i = int(num_trials / reduction_factor ** i)
+            min_i = int(numpy.ceil(min_resources * reduction_factor ** i))
+            bracket_budgets.append((n_i, min_i))
+
+        budgets.append(bracket_budgets)
+
+    return budgets
+
+
+class Hyperband(BaseAlgorithm):
+    """Hyperband
+
+    `Hyperparameter optimization [formulated] as a pure-exploration non-stochastic
+    infinite-armed bandit problem where a predefined resource like iterations, data samples,
+    or features is allocated to randomly sampled configurations.``
+
+    For more information on the algorithm,
+    see original paper at http://jmlr.org/papers/v18/16-558.html.
+
+    Li, Lisha et al. "Hyperband: A Novel Bandit-Based Approach to Hyperparameter Optimization"
+    Journal of Machine Learning Research, 18:1-52, 2018.
+
+    Parameters
+    ----------
+    space: `orion.algo.space.Space`
+        Optimisation space with priors for each dimension.
+    seed: None, int or sequence of int
+        Seed for the random number generator used to sample new trials.
+        Default: ``None``
+
+    """
+
+    def __init__(self, space, seed=None, repeat=numpy.inf):
+        self.brackets = []
+        super(Hyperband, self).__init__(space, seed=seed, repeat=numpy.inf)
+
+        self.trial_info = {}  # Stores Trial -> Bracket
+
+        try:
+            fidelity_index = self.fidelity_index
+        except IndexError:
+            raise RuntimeError(SPACE_ERROR)
+
+        fidelity_dim = space.values()[fidelity_index]
+
+        # min_resources = fidelity_dim.low
+        self.max_resources = fidelity_dim.high
+        self.reduction_factor = fidelity_dim.base
+
+        if self.reduction_factor < 2:
+            raise AttributeError("Reduction factor for Hyperband needs to be at least 2.")
+
+        self.repeat = repeat
+        self.execution_times = 0
+
+        self.budgets = compute_budgets(self.max_resources, self.reduction_factor)
+
+        self.brackets = [
+            Bracket(self, bracket_budgets)
+            for bracket_budgets in self.budgets
+        ]
+
+        self.seed_rng(seed)
+
+    def sample(self, num, bracket, buffer=10):
+        """Sample new points from bracket"""
+        samples = self.space.sample(num * buffer, seed=bracket.seed)
+        i = 0
+        points = []
+        while len(points) < num and i < num * buffer:
+            point = samples[i]
+            if self.get_id(point) not in self.trial_info:
+                point = list(point)
+                point[self.fidelity_index] = bracket.rungs[0]['resources']
+                points.append(tuple(point))
+            i += 1
+
+        if not points:
+            raise RuntimeError(
+                'Hyperband keeps sampling already existing points. This should not happen, '
+                'please report this error to '
+                'https://github.com/bouthilx/orion.algo.hyperband/issues')
```

Comment:
```suggestion
                'https://github.com/Epistimio/orion/issues')
```

Reply:
> Documentation will also need to be updated: https://orion.readthedocs.io/en/latest/user/algorithms.html#included-algorithms
> 
> (here https://github.com/Epistimio/orion/blob/master/docs/src/user/algorithms.rst)

Will add doc for hyperband later when we get some other issues done which may affect how to write the doc.